src/types/redis_zset.cc

```diff
@@ -71,19 +72,35 @@ rocksdb::Status ZSet::Add(const Slice &user_key, uint8_t flags, std::vector<Memb
     }
     added_member_keys.insert(member_key);
 
+    bool lt = (flags & kZSetLT) != 0;
+    bool gt = (flags & kZSetGT) != 0;
+
     if (metadata.size > 0) {
+      if (flags & kZSetNX) {
```

Comment:
I think it's not a correct condition. Because if there is a non-empty sorted set and you try to add a new member WITH the option `NX`, `kvrocks` doesn't add this element but `Redis` does. The root cause is that if the flag `NX` is set => then for loop does nothing because every time `continue` is executed (of course, if there is at least one element in the sorted set).

For example:

`Kvrocks`
```
[yaroslav@localhost redis]$ ./src/redis-cli -p 6666
127.0.0.1:6666> zadd ss 1 a 2 b 3 c
(integer) 3
127.0.0.1:6666> zrange ss 0 100
1) "a"
2) "b"
3) "c"
127.0.0.1:6666> zadd ss NX 4 d
(integer) 0     (<-- NOT added but this element is NEW)
127.0.0.1:6666> zrange ss 0 100
1) "a"
2) "b"
3) "c"
```
Compared to `Redis`:

```
[yaroslav@localhost redis]$ ./src/redis-cli
127.0.0.1:6379> zadd ss 1 a 2 b 3 c
(integer) 3
127.0.0.1:6379> zrange ss 0 100
1) "a"
2) "b"
3) "c"
127.0.0.1:6379> zadd ss NX 4 d
(integer) 1 (<-- ADDED)
127.0.0.1:6379> zrange ss 0 100
1) "a"
2) "b"
3) "c"
4) "d"
```